packages/validator/__tests__/validator.test.js

```diff
@@ -0,0 +1,797 @@
+      {
+        "name": "correct",
+        "operationParameter": {
+          "name": "operation",
+          "in": "header"
+        }
+      }
+    ]
+  },
+  {
+    "extra": "extra",
+    "fixedOperationParameters": [
+      {
+        "operationParameter": {
+          "in": "none"
+        },
+        "value": "param value"
+      }
+    ]
+  }
+]`;
+
+/* OIS SPECS */
+const specs19 = `{
+"oisFormat": "1.10.0",
+"title": "myOisTitle",
+"version": "1.2.3",
+"apiSpecifications": ${specs1},
+"endpoints": ${specs17}
+}`;
+
+const specs20 = `{
+"oisFormat": "1..0",
+"title": "myOisTitle",
+"version": "1.0",
+"apiSpecifications": ${specs15},
+"endpoints": ${specs17}
+}`;
+
+function formattingMessage(paramPath, error = false) {
+  return { level: error ? 'error' : 'warning', message: `${paramPath} is not formatted correctly` };
+}
+
+function keyFormattingMessage(key, paramPath) {
+  return { level: 'error', message: `Key ${key} in ${paramPath} is formatted incorrectly` };
+}
+
+function sizeExceededMessage(paramPath, maxSize) {
+  return { level: 'error', message: `${paramPath} must contain ${maxSize} or less items` };
+}
+
+function missingParamMessage(param) {
+  return { level: 'error', message: `Missing parameter ${param}` };
+}
+
+function extraFieldMessage(param) {
+  return { level: 'warning', message: `Extra field: ${param}` };
+}
+
+function conditionNotMetMessage(paramPath, param) {
+  return { level: 'error', message: `Condition in ${paramPath} is not met with ${param}` };
+}
+
+describe('validator', () => {
+  describe('api specs', () => {
+    test('valid configuration', () => {
+      expect(validator.isApiSpecsValid(specs1)).toMatchObject({ valid: true, messages: [] });
+      expect(validator.isApiSpecsValid(specs2)).toMatchObject({ valid: true, messages: [] });
+    });
+
+    test('invalid formatting', () => {
+      expect(validator.isApiSpecsValid(specs3)).toMatchObject({
+        valid: true,
+        messages: [formattingMessage('servers[0].url')],
+      });
+      expect(validator.isApiSpecsValid(specs4)).toMatchObject({
+        valid: true,
+        messages: [
+          formattingMessage('servers[0].url'),
+          keyFormattingMessage('myPath', 'paths.myPath'),
+          keyFormattingMessage('get ', 'paths.myPath.get '),
+          formattingMessage('paths.myPath.get .parameters[0].in'),
+          keyFormattingMessage('mySecurityScheme ', 'components.securitySchemes.mySecurityScheme '),
+          formattingMessage('components.securitySchemes.mySecurityScheme .in'),
+          keyFormattingMessage('mySecurityScheme ', 'security.mySecurityScheme ')
+        ],
+      });
+    });
+
+    test('multiple array items', () => {
+      expect(validator.isApiSpecsValid(specs5)).toMatchObject({ valid: true, messages: [] });
+      expect(validator.isApiSpecsValid(specs6)).toMatchObject({
+        valid: false,
+        messages: [sizeExceededMessage('servers', 1)],
+      });
+    });
+
+    test('edge cases', () => {
```

Comment:
Renamed in 5e852ff, I hope this would be enough, each group of tests is focused on one specific aspect of validator, which can be described using only few words, something more descriptive would probably mean writing out what exactly it is testing, which would take multiple lines and imho make it not easy to understand what the section is testing at first glance.

Reply:
Renames look good to me